app/src/main/java/com/example/fragmentassg2/FragmentPassword.kt

```diff
@@ -20,33 +21,54 @@ class FragmentPassword : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        tvFor?.setOnClickListener(this)
+        tvEnter?.setOnClickListener(this)
+        btnProceed?.setOnClickListener(this)
+        btnBack?.setOnClickListener(this)
+    }
 
-        tvFor.setOnClickListener(){
-            Toast.makeText(context,"Forgot Password",Toast.LENGTH_SHORT).show()
-        }
 
-        tvEnter.setOnClickListener(){
-            Toast.makeText(context,"Enter Password",Toast.LENGTH_SHORT).show()
-        }
+    override fun onClick(v: View?) {
+        val mobile= arguments?.getString("key")
 
-        tvGreat.setOnClickListener(){
-            Toast.makeText(context,"Great to see you again",Toast.LENGTH_SHORT).show()
-        }
+        when(v){
 
-        btnProceed.setOnClickListener(){
-            val pass:String="1234"
-            if(editPassword.text.toString()==pass){
-                val fragment= FragmentOTP()
-                val fragmentManager=activity!!.supportFragmentManager
-                val fragmentTransaction=fragmentManager.beginTransaction()
-                fragmentTransaction.setCustomAnimations(R.anim.enter_right_to_left,R.anim.exit_right_to_left,R.anim.enter_left_to_right,R.anim.exit_left_to_right)
-                fragmentTransaction.replace(R.id.root_layout,fragment)
-                fragmentTransaction.addToBackStack(null)
-                fragmentTransaction.commit()
-            }
-            else
-                Toast.makeText(context,"wrong",Toast.LENGTH_SHORT).show()
+            tvFor->Toast.makeText(context,getString(R.string.forgotpass),Toast.LENGTH_SHORT).show()
+            tvEnter->Toast.makeText(context,getString(R.string.pass),Toast.LENGTH_SHORT).show()
 
+
+            btnProceed->{
+                val pass="1234"
+                if(editPassword.editText?.text.toString()==pass){
+                    val fragmentOtp= FragmentOTP()
+                    val fragmentManager=activity!!.supportFragmentManager
+                    val fragmentTransaction=fragmentManager.beginTransaction()
+
+                    val bundle=Bundle()
+                    bundle.putString("key",mobile)
+                    fragmentOtp.arguments=bundle
+
+                    fragmentTransaction.setCustomAnimations(R.anim.enter_right_to_left,R.anim.exit_right_to_left,R.anim.enter_left_to_right,R.anim.exit_left_to_right)
+                    fragmentTransaction.replace(R.id.root_layout,fragmentOtp)
+                    fragmentTransaction.addToBackStack(null)
+                    fragmentTransaction.commit()
+                }
+                else
+                    Toast.makeText(context,"wrong",Toast.LENGTH_SHORT).show()
+            }
+            btnBack->{
+                replaceFragment()
```

Comment:
@raghav32 it should be in activity and not in fragment. We can talk about how to implement this